cns/restserver/ipam.go

```diff
@@ -34,6 +35,30 @@ const (
 	InterfaceName     = "eth0"
 )
 
+func (mcs *multicontainerstatus) String() string {
+	var unsuccessfulStatuses []string
+	for ncID, status := range *mcs {
+
+		ncStatus := status.CreateNetworkContainerRequest.NCStatus
+		if ncStatus != v1alpha.NCUpdateSuccess {
+			unsuccessfulStatus := fmt.Sprintf("Expected status for NC %s to be %s but got %s", ncID, string(v1alpha.NCUpdateSuccess), string(ncStatus))
+			unsuccessfulStatuses = append(unsuccessfulStatuses, unsuccessfulStatus)
+		}
+	}
+
+	return strings.Join(unsuccessfulStatuses, "\n")
+}
+
+func (mcs *multicontainerstatus) AnyFailed() bool {
+	for _, status := range *mcs {
+		if status.CreateNetworkContainerRequest.NCStatus != v1alpha.NCUpdateSuccess {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
This method won't be needed as if you get 0 unsuccessfulErrorStatus Back you can infer that all NCs are in success mode.

Reply:
Removed